### PR TITLE
fix: async call to log

### LINKED ISSUE #209: No longer shows error outputs on building
### Bug description

There is no more error outputs in the logging.

### Operating system

mac m1

### Version

next (development version)

### Steps to reproduce

1. Select ISO (it will fail as this is a known bug)
2. Check log output
3. It wont show the errored output:

```
Build Image Log --------
Image:  quay.io/podai/test
Type:   iso
Folder: /Users/cdrage/bootc11232
----------
{
  "name": "test-bootc-image-builder",
  "Image": "quay.io/centos-bootc/bootc-image-builder:latest-1710916056",
  "Tty": true,
  "HostConfig": {
    "Privileged": true,
    "SecurityOpt": [
      "label=type:unconfined_t"
    ],
    "Binds": [
      "/Users/cdrage/bootc11232:/output/",
      "/var/lib/containers/storage:/var/lib/containers/storage"
    ]
  },
  "Labels": {
    "bootc.image.builder": "true",
    "bootc.build.image.location": "/Users/cdrage/bootc11232/bootiso/disk.iso",
    "bootc.build.type": "iso"
  },
  "Cmd": [
    "quay.io/podai/test:latest",
    "--type",
    "iso",
    "--target-arch",
    "amd64",
    "--output",
    "/output/",
    "--local"
  ]
}
----------
```

Most likely because we no longer put it in 'finally' statement: https://github.com/containers/podman-desktop-extension-bootc/pull/205/files#diff-08ed5ed5987d1721bd2408890fd2c46bbbbeabf06ec0f7dff72a58313185e96fL206

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Fixes async call that stopped logging.

### Screenshot / video of UI

<!-- If this PR is changing UI, please include 
N/A

### What issues does this PR fix or reference?

Fixes #209.

### How to test this PR?

Build, confirm log file is correct now.

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
still missing unit tests

**Review by cdrage — CHANGES_REQUESTED** [review]
I still get no error output, does not fix the issue.

https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/1d80d909-7104-4453-b2ef-258059a45d97


You can also replicate it and see how it is still not getting all the container log output by:
1. Build
2. QUICKLY switch to container building
3. See that it's outputting a lot of information (such as pulling the container)>
4. It does not appear in logs

Happens with both .iso and .qcow2 selection.


https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/7e0bd84d-cdd1-43ba-802a-7ebd43819f35

**Comment by cdrage** [comment]
https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/2b8906cc-b206-4152-9305-8e70d7c0243a

I can confirm it now works, thank you 👍

**Review by cdrage — APPROVED** [review]

File: packages/backend/src/build-disk-image.ts [hunk 1]
```diff
@@ -176,13 +176,14 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History): P
 
           // Step 3.1 Since we have started the container, we can now go get the logs
           await logContainer(build.engineId, containerId, progress, data => {
-            async () => {
+            // update the log file asyncronously
+            void (async () => {
               try {
```
Comment (benoitf) [inline packages/backend/src/build-disk-image.ts:181]: AFAIK you should not use the void keyword there
Just call the function without await but catch error

Comment (deboer-tim) [reply packages/backend/src/build-disk-image.ts:182]: You're right. Removed void and rebased on main.

**Review by benoitf — COMMENTED** [review]
looks like the call should be

```ts
await logContainer(build.engineId, containerId, progress, data => {
 // update the log file asynchronously
fs.promises.appendFile(logPath, data).catch((error:unknown) => { 
console.debug('Could not write bootc build log: ', e);
});
});
```

I don't see why wrap add an extra async call.

**Comment by deboer-tim** [comment]
> looks like the call should be
> 
> ```ts
> await logContainer(build.engineId, containerId, progress, data => {
>  // update the log file asynchronously
> fs.promises.appendFile(logPath, data).catch((error:unknown) => { 
> console.debug('Could not write bootc build log: ', e);
> });
> });
> ```
> 
> I don't see why wrap add an extra async call.

Ahhh, that's what you meant. Sorry for being dense. It started with a 'traditional' try { } catch, which the linter flags as needed await, which caused me to go down the rathole of trying to make it happy while still working. Re-pushed with your suggestion, agree it is much better, linter happy, and most importantly it works.

**Comment by benoitf** [comment]
yes I was not clear enough

**Pull request merged**